### ISSUE TITLE
BackdropFilter blurs everything on the screen when there is a platformview on iOS (Regression of flutter 1.17.0)

### ISSUE THREAD

**Issue by Tetracyl** [flutter/flutter#80766]
## Issue
When I use the BackdropFilter, it works perfectly fine on Android and blurs out the background but not the button. However, when I use it on iOS it blurs everything.

Android | iOS
------------ | -------------
![image](https://user-images.githubusercontent.com/29833473/115339696-2c762180-a15a-11eb-8231-4ce629b68ef6.png) | ![image0](https://user-images.githubusercontent.com/29833473/115339790-4dd70d80-a15a-11eb-96d1-fc605bd4a4a1.png)

## Steps to Reproduce

1. Create a stack with some widgets
2. Create a backdrop filter with a blur inside the stack
3. Set a button as the child of the backdrop filter

### Code
I've removed some of the unnecessary code to make it easier to read.

<details> 
<summary>code snippet</summary> 

```dart
class GuestView extends StatefulWidget {
  @override
  _GuestViewState createState() => _GuestViewState();
}

class _GuestViewState extends State<GuestView> {
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 5.0),
      child: Stack(
        children: [
          SingleChildScrollView(
            physics: NeverScrollableScrollPhysics(),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisAlignment: MainAxisAlignment.start,
              children: <Widget>[
                Icon(
                  Icons.account_circle,
                  size: 150,
                  color: Color(0xFF40444b),
                ),
                Padding(
                  padding: const EdgeInsets.only(bottom: 10.0),
                  child: RichText(
                    ...
                  ),
                ),
                AccountBadges(
                    ...
                ),
                Divider(
                  color: Theme.of(context).dividerColor,
                  thickness: 0.2,
                ),
                AccountButtonsWidget(account: null),
              ],
            ),
          ),
          BackdropFilter(
            filter: ImageFilter.blur(sigmaX: 6, sigmaY: 6),
            child: Center(
              child: Container(
                decoration: BoxDecoration(
                  color: Colors.white,
                  borderRadius: BorderRadius.only(
                      topLeft: Radius.circular(10),
                      topRight: Radius.circular(10),
                      bottomLeft: Radius.circular(10),
                      bottomRight: Radius.circular(10)),
                  boxShadow: [
                    BoxShadow(
                      color: Colors.black.withOpacity(0.5),
                      spreadRadius: 6,
                      blurRadius: 5,
                    ),
                  ],
                ),
                child: MaterialFlatButton(
                  ...
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
``` 

</details>

## Logs

<details>
<summary>Logs</summary>

Windows:
```
[√] Flutter (Channel stable, 2.0.5, on Microsoft Windows [Version 10.0.19042.928], locale en-US)
    • Flutter version 2.0.5 at C:\Tools\flutter
    • Framework revision adc687823a (4 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\seif\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java       
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2020.3.2
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.55.2)
    • VS Code at C:\Users\seif\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.21.0

[√] Connected device (3 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 89.0.4389.128
    • Edge (web)                  • edge          • web-javascript • Microsoft Edge 89.0.774.77
```
Mac:
```
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-US)
    • Flutter version 2.0.5 at /Users/seifabdelaziz/Development/flutter
    • Framework revision adc687823a (4 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/seifabdelaziz/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[!] Connected device
    ! No devices available

```
</details>

**Comment by TahaTesser** [comment]
Hi @Tetracyl 
Thanks for filing the issue, It might have something to do with AdMob you've, which uses platform view

Can you please reproduce the issue without using a plugin? 

Your code sample is incomplete so I tried to create a complete runnable code samplem isse doesn't reproduce on iOS or Androod

<details> 
<summary>code sample</summary> 

```dart
import 'dart:ui';

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: GuestView(),
    );
  }
}

class GuestView extends StatefulWidget {
  @override
  _GuestViewState createState() => _GuestViewState();
}

class _GuestViewState extends State<GuestView> {
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 5.0),
      child: Stack(
        children: [
          FlutterLogo(
            size: 300,
          ),
          BackdropFilter(
            filter: ImageFilter.blur(sigmaX: 6, sigmaY: 6),
            child: Center(
              child: Container(
                decoration: BoxDecoration(
                  color: Colors.white,
                  borderRadius: BorderRadius.only(
                      topLeft: Radius.circular(10),
                      topRight: Radius.circular(10),
                      bottomLeft: Radius.circular(10),
                      bottomRight: Radius.circular(10)),
                  boxShadow: [
                    BoxShadow(
                      color: Colors.black.withOpacity(0.5),
                      spreadRadius: 6,
                      blurRadius: 5,
                    ),
                  ],
                ),
                child: Container(
                  height: 100,
                  width: double.infinity,
                  child: Center(child: Text("Discord")),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

``` 

</details>

Also using the official sample has no issue

<details> 
<summary>flutter doctor -v</summary> 

```bash
import 'dart:ui' as ui;

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: GuestView(),
    );
  }
}

class GuestView extends StatefulWidget {
  @override
  _GuestViewState createState() => _GuestViewState();
}

class _GuestViewState extends State<GuestView> {
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 5.0),
      child: Stack(
        fit: StackFit.expand,
        children: <Widget>[
          Text('0' * 10000),
          Center(
            child: ClipRect(
              // <-- clips to the 200x200 [Container] below
              child: BackdropFilter(
                filter: ui.ImageFilter.blur(
                  sigmaX: 5.0,
                  sigmaY: 5.0,
                ),
                child: Container(
                  alignment: Alignment.center,
                  width: 200.0,
                  height: 200.0,
                  child: Text('Hello World'),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
```
	
</details>


Can you please provide your a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) 
Thank you

**Comment by Tetracyl** [comment]
You're right, it's a problem with the platform view. I could not reproduce the issue with your code sample as well, however, once I added the banner ad the issue reappeared. 

| Without PlatformVIew  | With Platform View |
| ------------- | ------------- |
| ![Simulator Screen Shot - iPhone SE (2nd generation) - 2021-04-20 at 06 37 08](https://user-images.githubusercontent.com/29833473/115405369-088afe00-a1a3-11eb-9d32-3b559c5f457d.png) | ![Simulator Screen Shot - iPhone SE (2nd generation) - 2021-04-20 at 06 37 31](https://user-images.githubusercontent.com/29833473/115405446-16d91a00-a1a3-11eb-92ba-5945185c8c7c.png) |

Here is the reproducable code sample:
```dart
import 'dart:ui';

import 'package:flutter/material.dart';
import 'package:native_admob_flutter/native_admob_flutter.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await MobileAds.initialize();
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: GuestView(),
    );
  }
}

class GuestView extends StatefulWidget {
  @override
  _GuestViewState createState() => _GuestViewState();
}

class _GuestViewState extends State<GuestView> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Home'),
        centerTitle: true,
      ),
      body: Column(
        children: [
          Expanded(
            child: Padding(
              padding:
                  const EdgeInsets.symmetric(horizontal: 20.0, vertical: 5.0),
              child: Stack(
                children: [
                  FlutterLogo(
                    size: 300,
                  ),
                  BackdropFilter(
                    filter: ImageFilter.blur(sigmaX: 8, sigmaY: 8),
                    child: Center(
                      child: Container(
                        decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: BorderRadius.only(
                              topLeft: Radius.circular(10),
                              topRight: Radius.circular(10),
                              bottomLeft: Radius.circular(10),
                              bottomRight: Radius.circular(10)),
                          boxShadow: [
                            BoxShadow(
                              color: Colors.black.withOpacity(0.5),
                              spreadRadius: 6,
                              blurRadius: 5,
                            ),
                          ],
                        ),
                        child: Container(
                          height: 100,
                          width: double.infinity,
                          child: Center(child: Text("Discord")),
                        ),
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
          BannerAd(),
        ],
      ),
    );
  }
}
```
Note: You'll have to import the `native_admob_flutter` package.

**Comment by TahaTesser** [comment]
Hi @Tetracyl 
Thanks for the details

In order to reproduce the issue using minimal steps, we can use `webview_flutter` instead

This issue is closely related to https://github.com/flutter/flutter/issues/43902

Where blur is not applied on platform view but here in the presence of platform view, it blurs everything else on the screen. you can see the same result in my old comment https://github.com/flutter/flutter/issues/43902#issuecomment-637357435.
When we try to blur platform view on iOS, it doesn't work but it makes everything else blurry instead https://github.com/flutter/flutter/issues/43902#issuecomment-731228962


<details> 
<summary>complete runnable minimal code sample</summary> 

```dart
import 'dart:io';
import 'dart:ui';
import 'package:webview_flutter/webview_flutter.dart';
import 'package:flutter/material.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: GuestView(),
    );
  }
}

class GuestView extends StatefulWidget {
  @override
  _GuestViewState createState() => _GuestViewState();
}

class _GuestViewState extends State<GuestView> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          Expanded(
            child: Padding(
              padding:
                  const EdgeInsets.symmetric(horizontal: 20.0, vertical: 5.0),
              child: Stack(
                children: [
                  FlutterLogo(
                    size: 300,
                  ),
                  BackdropFilter(
                    filter: ImageFilter.blur(sigmaX: 8, sigmaY: 8),
                    child: Center(
                      child: Container(
                        decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: BorderRadius.only(
                              topLeft: Radius.circular(10),
                              topRight: Radius.circular(10),
                              bottomLeft: Radius.circular(10),
                              bottomRight: Radius.circular(10)),
                          boxShadow: [
                            BoxShadow(
                              color: Colors.black.withOpacity(0.5),
                              spreadRadius: 6,
                              blurRadius: 5,
                            ),
                          ],
                        ),
                        child: Container(
                          height: 100,
                          width: double.infinity,
                          child: Center(child: Text("Discord")),
                        ),
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
          Container(height: 300, child: WebViewExample()),
        ],
      ),
    );
  }
}

class WebViewExample extends StatefulWidget {
  @override
  WebViewExampleState createState() => WebViewExampleState();
}

class WebViewExampleState extends State<WebViewExample> {
  @override
  void initState() {
    super.initState();
    // Enable hybrid composition.
    if (Platform.isAndroid) WebView.platform = SurfaceAndroidWebView();
  }

  @override
  Widget build(BuildContext context) {
    return WebView(
      initialUrl: 'https://flutter.dev',
    );
  }
}

``` 

</details>


### Preview
| iOS (with webview) | iOS (without webview) |
| --------------- | --------------- |
<img src="https://user-images.githubusercontent.com/48603081/115517277-a65bf700-a28f-11eb-9910-a1d6e39b7493.png" height="450" /> | <img src="https://user-images.githubusercontent.com/48603081/115517284-a78d2400-a28f-11eb-9bd7-cfa40d500ab0.png" height="450" /> |



<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale
    en-GB)
    • Flutter version 2.0.5 at /Users/tahatesser/Code/flutter_stable
    • Framework revision adc687823a (5 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 55.1.5
    • Dart plugin version 211.6693.108

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 83060656-28E7-44CD-801E-B11EC3EC89BA • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 89.0.4389.128

• No issues found!
```

```bash
[✓] Flutter (Channel master, 2.2.0-11.0.pre.187, on macOS 11.2.3 20D91
    darwin-x64, locale en-GB)
    • Flutter version 2.2.0-11.0.pre.187 at
      /Users/tahatesser/Code/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5526dcc24b (2 hours ago), 2021-04-20 22:34:02 -0700
    • Engine revision 610fd039ae
    • Dart version 2.14.0 (build 2.14.0-18.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 55.1.5
    • Dart plugin version 211.6693.108

[!] IntelliJ IDEA Community Edition (version unknown)
    ✗ Invalid IntelliJ version number.

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 83060656-28E7-44CD-801E-B11EC3EC89BA • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 89.0.4389.128

! Doctor found issues in 1 category.
```
	
</details>

I think when https://github.com/flutter/flutter/issues/43902 is fixed, this should be fixed too

**Comment by bdlukaa** [comment]
Try wrapping your `BackdropFilter` in a `ClipRect`. Worked for me.

**Comment by Tetracyl** [comment]
Didn't work for me.

<img width="348" alt="image" src="https://user-images.githubusercontent.com/29833473/115634179-eb971d80-a2bd-11eb-95f4-c05a07cc4e5d.png">

**Comment by chinmaygarde** [comment]
cc @blasten. The behavior on Android should become identical to that on iOS if the application were to opt into using hybrid composition right? Once that happens, supporting the background filter in the mutator stack (we don't do this today) ought to be the next step.

**Comment by flar** [comment]
Similar to https://github.com/flutter/flutter/issues/56789 ?

**Comment by blasten** [comment]
This actually seems to be a bug in the composition. It looks like having a platform view causes `BackdropFilter` to affect the entire context of the layer that the engine painted below the platform view. 

@chinmaygarde - This doesn't affect the mutator stack since the part that is affected isn't the platform view, but the Flutter UI.

**Comment by kenthinson** [comment]
@chinmaygarde I am able to reproduce this bug on MacOS as well as iOS.

**Comment by QuentinSc** [comment]
My app still not public because of this..

**Comment by cyanglaz** [comment]
This is a regression of v1.17.0 based on https://github.com/flutter/flutter/issues/56789#issue-615247489

**Comment by cyanglaz** [comment]
Bisect says https://github.com/flutter/flutter/pull/53200 is the offending commit. if the leaf_nodes_canvas is not updated, the issue doesn't occur. (For example, trying to comment out https://github.com/flutter/engine/blob/master/flow/layers/platform_view_layer.cc#L40 and the issue won't happen)
I think it might related to unobstructed platform views(https://github.com/flutter/flutter/commit/f5d25da83ceadffea850cae369f95d3fc1fde85f). I can't confirm the engine commit as I'm not able to build local engine that old.

I will further investigate, @blasten Let me know if you have any hunch.

**Comment by cyanglaz** [comment]
Moving [beginRecording](`https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L302`) for the canvas to paint traversal will fix the issue.
@blasten I know it is a long time ago but did you remember a particular reason the `beginRecording` has to happen in preroll?

**Comment by cyanglaz** [comment]
After asking the skia team, I had more understanding on how backdrop filter works and now I understand the root cause of this issue.

Backdrop filter takes the pixels on the original canvas and filter them, then replace the pixels on the original canvas.
backdrop_filter_layer of Flutter engine, we [apply the backdrop filter layer to all the canvas](https://github.com/flutter/engine/blob/master/flow/layers/layer.cc#L69)
Because with platform view, there is an additional picture recorder drawing on top of the original canvas. This picture recorder [started recording in preroll traversal](https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L302), which caused it to contain the backdrop filter layer.
Then when this picture recorder's picture is [drawn on the original canvas](https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L547), it filtered all the existing content on the original canvas, which leads to this bug.

To sum up, in this bug, 2 backdrop filter has happened, that's why both background and the text box are filtered. 

Still looking for a solution.

**Comment by cyanglaz** [comment]
I have drafted a document explaining the issue and proposed a potential solution. https://docs.google.com/document/d/1t_offKOgkuVzvB3fqGTfcltovb-yxqV6tbuP1qIBqVI/edit?usp=sharing
Please feel free to comment on it. Once the solution is finalized, I will start working on the implementation.
